tools/flang/main.cpp

```diff
@@ -0,0 +1,81 @@
+#include "clang/Basic/DiagnosticIDs.h"
+#include "clang/Basic/DiagnosticOptions.h"
+#include "clang/Driver/Compilation.h"
+#include "clang/Driver/Driver.h"
+#include "clang/Driver/Job.h"
+#include "clang/Driver/ToolChain.h"
+#include "clang/Frontend/CompilerInvocation.h"
+#include "clang/Frontend/TextDiagnosticPrinter.h"
+#include "clang/FrontendTool/Utils.h"
+#include "llvm/ADT/ArrayRef.h"
+#include "llvm/ADT/IntrusiveRefCntPtr.h"
+#include "llvm/ADT/SmallString.h"
+#include "llvm/ADT/SmallVector.h"
+#include "llvm/ADT/StringRef.h"
+#include "llvm/Option/ArgList.h"
+#include "llvm/Option/Option.h"
+#include "llvm/Support/CommandLine.h"
+#include "llvm/Support/FileSystem.h"
+#include "llvm/Support/Host.h"
+#include "llvm/Support/InitLLVM.h"
+#include "llvm/Support/Path.h"
+#include "llvm/Support/Program.h"
+#include "llvm/Support/VirtualFileSystem.h"
+
+#include <memory>
+#include <set>
+#include <stdlib.h>
+#include <system_error>
+
+std::string GetExecutablePath(const char *Argv0) {
+  // This just needs to be some symbol in the binary; C++ doesn't
+  // allow taking the address of ::main however.
+  void *P = (void *)(intptr_t)GetExecutablePath;
+  return llvm::sys::fs::getMainExecutable(Argv0, P);
+}
+
+// This lets us create the DiagnosticsEngine with a properly-filled-out
+// DiagnosticOptions instance
+static clang::DiagnosticOptions *CreateAndPopulateDiagOpts(
+    llvm::ArrayRef<const char *> argv) {
+  auto *DiagOpts = new clang::DiagnosticOptions;
+  unsigned MissingArgIndex, MissingArgCount;
```

Comment:
`unsigned MissingArgIndex{}, MissingArgCount{};` 
This avoids having to disable the warning above. I realise though that these two values are set to 0 inside ParseArgs straight away so this isn't actually a bug.

Reply:
I believe that is done in the CMakeFileLists.txt by removing almost all -wno

Reply:
I agree with David here. If you can reasonably change the code to avoid a warning then we should do that rather than disabling the warning. Is there a good reason to not do this?

Surely as compiler writers we don't like people who ignore the nice warnings we have striven hard to implement?! :-)

Reply:
I believe this is fixed!